x/evm/keeper/vote_handler.go

```diff
@@ -227,17 +226,7 @@ func (v voteHandler) handleContractCall(ctx sdk.Context, event types.Event) erro
 		return err
 	}
 
-	// this is still here only because routing to modules other than EVM can be
-	// expensive. this is currently called by the last voter, and the extra gas
-	// may cause trouble for gas estimation.
-	if !msg.Recipient.Chain.IsFrom(types.ModuleName) {
-		return nil
-	}
-
-	// try routing the message
-	_ = utils.RunCached(ctx, v.keeper, func(ctx sdk.Context) (struct{}, error) {
-		return struct{}{}, v.nexus.RouteMessage(ctx, msg.ID)
-	})
+	funcs.MustNoErr(v.nexus.EnqueueRouteMessage(ctx, msg.ID))
```

Comment:
set event as completed here instead of the workaround in evm handleMessages? Fine to do this in a separate PR as well

Reply:
It's a separate PR with separate JIRA ticket.